drivers/bluetooth/hci/userchan.c

```diff
@@ -257,14 +291,18 @@ SYS_INIT(bt_uc_init, POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE);
 
 static void cmd_bt_dev_found(char *argv, int offset)
 {
-	if (strncmp(&argv[offset], "hci", 3) || strlen(&argv[offset]) < 4) {
-		posix_print_error_and_exit("Error: Invalid Bluetooth device "
-					   "name '%s' (should be e.g. hci0)\n",
+	arg_found = true;
+	if (strncmp(&argv[offset], "hci", 3) == 0 && strlen(&argv[offset]) >= 4) {
+		/* It's a hciX string */
+		hci_socket = true;
+		bt_dev_index = strtol(&argv[offset + 3], NULL, 10);
```

Comment:
`strtol` may return values outside the range of an `int`

Reply:
The current implementation in main uses strol  ,  this is not something I changed. Sugfrstions are welcome

https://github.com/zephyrproject-rtos/zephyr/blob/d11c0a1664f6052e68c6eebd35aa5aba88e7c290/drivers/bluetooth/hci/userchan.c#L267

Reply:
One suggestion would be to store the result of `strtol` in a local `long` variable, verify that the value is within range of `bt_dev_index`, and then only then assign it. 
Alternatively use `atoi`

Arguably `bt_dev_index` should also be an unsigned value.